hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultPortableReader.java

```diff
@@ -756,7 +762,7 @@ private void validateNotMultiPosition(PortablePosition position) {
     private void validateType(PortablePosition position, FieldType expectedType) {
         FieldType returnedType = position.getType();
         if (position.getIndex() >= 0) {
-            returnedType = returnedType.getSingleType();
+            returnedType = returnedType != null ? returnedType.getSingleType() : null;
         }
```

Comment:
It was a possible null pointer in unexpected cases that should be guarded by the caller, but just in case.